bin/oneoff/list_s3_buckets

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env ruby
+
+# Inventory Code.org S3 buckets.
+
+require_relative '../../dashboard/config/environment'
+require 'aws-sdk-s3'
+require 'aws-sdk-cloudwatch'
+require 'ostruct'
+require 'csv'
+
+s3_client = Aws::S3::Client.new
+cloudwatch_client = Aws::CloudWatch::Client.new
+PRIMARY_REGION = 'us-east-1'.freeze
+ACTIVITY_WINDOW = 7.days     # Window of time within which bucket Metrics are collected.
+METRIC_GRANULARITY = 1.hour
+
+Bucket = Struct.new(
+  :name,
+  :creation_date,
+  :cloudformation, # Is the bucket provisioned via code (CloudFormation)?
+  :region,
+  :versioning,
+  :encryption,
+  :storage_size,
+  :object_count,
+  :bytes_uploaded,
+  :get_requests,
+  :put_requests,
+  :delete_requests
+)
+
+buckets = []
+
+s3_buckets = s3_client.list_buckets.buckets
+s3_buckets.each do |s3_bucket|
+  bucket = Bucket.new
+  bucket[:name] = s3_bucket.name
+  bucket[:creation_date] = s3_bucket.creation_date
+  begin
+    bucket[:cloudformation] = s3_client.
+      get_bucket_tagging(bucket: bucket[:name]).
+      tag_set.
+      any? {|tag| tag[:key].starts_with?('aws:cloudformation:')}
+  rescue Aws::S3::Errors::NoSuchTagSet => exception
+    puts exception.message
+    bucket[:cloudformation] = false
+  end
+  bucket[:region] = s3_client.get_bucket_location(bucket: bucket[:name]).location_constraint
+  bucket[:region] = PRIMARY_REGION if bucket[:region] == ''
+  bucket[:versioning] = s3_client.get_bucket_versioning(bucket: bucket[:name]).status
+  begin
+    bucket[:encryption] = s3_client.get_bucket_encryption(bucket: bucket[:name]).
+      server_side_encryption_configuration.
+      rules.
+      map {|rule| rule.apply_server_side_encryption_by_default.sse_algorithm}.
+      join(':')
+  rescue Aws::S3::Errors::ServerSideEncryptionConfigurationNotFoundError => exception
+    puts "IGNORING - #{exception.message}"
+  end
+
+  bucket[:storage_size] = Aws::CloudWatch::Metric.new(namespace: 'AWS/S3', name: 'BucketSizeBytes', client: cloudwatch_client).
```

Comment:
I plan to implement a `get_statistics` method to use for all of this repeated logic.